volume/volume.go

```diff
@@ -124,14 +124,15 @@ type MountPoint struct {
 
 // Setup sets up a mount point by either mounting the volume if it is
 // configured, or creating the source directory if supplied.
-func (m *MountPoint) Setup(mountLabel string, rootUID, rootGID int) (path string, err error) {
+func (m *MountPoint) Setup(prefix, mountLabel string, rootUID, rootGID int) (path string, err error) {
+	source := filepath.Join(prefix, m.Source)
 	defer func() {
 		if err == nil {
 			if label.RelabelNeeded(m.Mode) {
-				sourcePath, err := filepath.EvalSymlinks(m.Source)
+				sourcePath, err := filepath.EvalSymlinks(source)
```

Comment:
Not sure how well this will work.  If /host/etc/foobar is a symlink to /etc/foobar.new  Then won't the source end up causing issues?

Reply:
ah thanks to have pointed it out.  I needed to do the resolution in any case, not only for the relabelling part, otherwise symlinks would be resolved without the prefix.